services/121-service/src/payments/reconciliation/onafriq-reconciliation/onafriq-reconciliation.service.ts

```diff
@@ -119,36 +129,38 @@ export class OnafriqReconciliationService {
   }
 
   public async sendReconciliationReport(): Promise<number> {
-    const result = await this.generateAndSendReconciliationReportYesterday();
-    return result.length;
+    const programs = await this.programRepository.find();
+    let result = 0;
+    for (const program of programs) {
+      const report = await this.generateAndSendReconciliationReportYesterday(
+        program.id,
+      );
+      result += report.length;
+    }
+    return result;
   }
 
   public async generateAndSendReconciliationReportYesterday(
+    programId: number,
     toDate?: Date,
     fromDate?: Date,
   ): Promise<OnafriqReconciliationReport[]> {
```

Comment:
Make deconstructed object, especially considering programId is not in part of the function name